contracts/BC_fusion/StakeHub.sol

```diff
@@ -0,0 +1,672 @@
+    event DescriptionEdited(address indexed validator);
+    event Delegated(address indexed validator, address indexed delegator, uint256 bnbAmount);
+    event Undelegated(address indexed validator, address indexed delegator, uint256 bnbAmount);
+    event Redelegated(address indexed srcValidator, address indexed dstValidator, address indexed delegator, uint256 bnbAmount);
+    event ValidatorSlashed(address indexed validator, uint256 slashAmount, uint256 slashHeight, uint256 jailUntil, SlashType slashType);
+    event ValidatorJailed(address indexed validator);
+    event ValidatorUnjailed(address indexed validator);
+    event Claimed(address indexed validator, address indexed delegator, uint256 bnbAmount);
+    event SecurityDepositClaimed(address indexed validator, uint256 sharesAmount);
+    event StakingPaused();
+    event StakingResumed();
+    event paramChange(string key, bytes value);
+
+    /*----------------- modifiers -----------------*/
+    modifier onlyInitialized() {
+        require(_initialized != 0, "NOT_INITIALIZED");
+        _;
+    }
+
+    modifier validatorExist(address validator) {
+        require(validators[validator].poolModule != address(0), "VALIDATOR_NOT_EXIST");
+        _;
+    }
+
+    modifier validatorNotJailed(address validator) {
+        require(!validators[validator].jailed, "VALIDATOR_JAILED");
+        _;
+    }
+
+    modifier onlyOperator(address validator) {
+        require(validators[validator].poolModule != address(0), "VALIDATOR_NOT_EXIST");
+        if (msg.sender != validator) {
+            require(msg.sender == validators[validator].operatorAddress, "NOT_OPERATOR");
+        }
+        _;
+    }
+
+    modifier whenNotPaused() {
+        require(!_stakingPaused, "STAKE_STOPPED");
+        _;
+    }
+
+    /*----------------- init -----------------*/
+    function initialize() public {
+        transferGasLimit = INIT_TRANSFER_GAS_LIMIT;
+
+        minSelfDelegationBNB = INIT_MIN_SELF_DELEGATION_BNB;
+        minDelegationBNBChange = INIT_MIN_DELEGATION_BNB_CHANGE;
+        maxElectedValidators = INIT_MAX_ELECTED_VALIDATORS;
+        unbondPeriod = INIT_UNBOND_PERIOD;
+        downtimeSlashAmount = INIT_DOWNTIME_SLASH_AMOUNT;
+        doubleSignSlashAmount = INIT_DOUBLE_SIGN_SLASH_AMOUNT;
+        downtimeJailTime = INIT_DOWNTIME_JAIL_TIME;
+        doubleSignJailTime = INIT_DOUBLE_SIGN_JAIL_TIME;
+        maxEvidenceAge = INIT_MAX_EVIDENCE_AGE;
+
+        _initialized = 1;
+    }
+
+    /*----------------- external functions -----------------*/
+    function createValidator(address consensusAddress, bytes calldata voteAddress, bytes calldata blsProof, Commission calldata commission, Description calldata description) external payable onlyInitialized {
+        uint256 delegation = msg.value;
+        require(delegation >= minSelfDelegationBNB, "INVALID_SELF_DELEGATION");
+        require(validators[consensusAddress].poolModule == address(0), "ALREADY_VALIDATOR");
```

Comment:
I think it is better to use `operatorAddress` as a key.

Reply:
and use `operatorAddress` as the index of a validator.

Becuase we can use `operatorAddress` account to update the consensus key, because consensus key is running with node, it is often rotated.

Reply:
Done